.github/workflows/ci.yml

```diff
@@ -0,0 +1,144 @@
+name: CI
+
+on: [ push, pull_request, workflow_dispatch ]
+
+jobs:
+  CI:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ contains(matrix.os, 'macos') }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [ "Linux x64 (Ubuntu 20.04) - GCC 10" ]
+        os: [ ubuntu-20.04 ]
+        sanitize: [ false ]
+        build:
+          - { cc: gcc, cxx: g++, linker: ld }
+        include:
+          - name: "Linux x64 (Ubuntu 20.04) - Clang 10 with ASan and UBSan"
+            os: ubuntu-20.04
+            sanitize: true
+            build: { cc: clang-10, cxx: clang++-10, linker: ld.lld-10 }
+          - name: "macOS (10.15) - Xcode 12.3"
+            os: macos-10.15
+            sanitize: false
+            build: { cc: clang, cxx: clang++, linker: ld.lld }
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Update apt
+        if: contains(matrix.os, 'ubuntu')
+        env:
+          DEBIAN_FRONTEND: noninteractive
+        run: sudo -E apt-get update -qq -o Acquire::Retries=3
+
+      - name: Add libheif PPA
+        if: contains(matrix.os, 'ubuntu')
+        run: |
+          sudo add-apt-repository ppa:strukturag/libde265
+          sudo add-apt-repository ppa:strukturag/libheif
+
+      - name: Install Ubuntu dependencies
+        if: contains(matrix.os, 'ubuntu')
+        env:
+          DEBIAN_FRONTEND: noninteractive
+        run: |
+          sudo -E apt-get install --fix-missing -qq -o Acquire::Retries=3 \
+            gtk-doc-tools gobject-introspection \
+            python3-pip python3-setuptools python3-wheel \
+            libfftw3-dev libexif-dev libjpeg-turbo8-dev \
+            libpng-dev libwebp-dev libtiff5-dev \
+            libheif-dev libexpat1-dev libcfitsio-dev \
+            libmatio-dev libnifti-dev liborc-0.4-dev \
+            liblcms2-dev libpoppler-glib-dev librsvg2-dev \
+            libgif-dev libopenexr-dev libpango1.0-dev \
+            libgsf-1-dev libopenslide-dev libffi-dev
+
+      - name: Install macOS dependencies
+        if: contains(matrix.os, 'macos')
+        run: |
+          brew install \
+            autoconf automake libtool \
```

Comment:
I think you don't need the end of line backslash, though perhaps it's clearer to have it.

Reply:
Yes, I thought the `run: |` command needs escaping where `run:` (without `|`) doesn't, but I'm not sure.

Reply:
Could use `>` to fold (replace newline with space):
```yaml
  run: >
    brew install
      autoconf automake libtool
```

Reply:
I tried to use the fold sytnax, see: https://github.com/kleisauke/libvips/commit/1ca8e00165a7ad496c9f014b656bb4b27d5197e2. Unfortunally, [this fails](https://github.com/kleisauke/libvips/actions/runs/551502470) since the fold syntax requires the same indentation for all lines in order to turn newlines into spaces.

Somehow the normal syntax (`run:`, without `|` or `>`), does seem to work just fine, see: https://github.com/libvips/libvips/pull/1986/commits/b93db2f2bb7795f49f3421f5bbfa33fca273650c (not sure if this is a bug or a feature).